### PR TITLE
Added BARTScore

### LINKED ISSUE #71: New Metric: Add BARTScore
BARTScore was published in NIPS'21. We can add it to jury with minimum requirement set.

Repo: https://github.com/neulab/BARTScore
Paper: https://arxiv.org/abs/2106.11520

### PR DESCRIPTION
**Pull request by Sophylax**
Closes #71.

### CODE REVIEW

File: jury/metrics/bartscore/bartscore_for_language_generation.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            license=_LICENSE,
+        )
+
+    def _compute_single_pred_single_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        batch_size: int = 4,
+        segment_scores: bool = False,
```
Comment (devrimcavusoglu) [inline jury/metrics/bartscore/bartscore_for_language_generation.py:194]: very nice touch with `segment_scores` :heart:

File: tests/jury/metrics/test_bartscore.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@pytest.fixture
+@get_expected_output(prefix="metrics")
+def output_multiple_pred_multiple_ref():
+    return output_multiple_pred_multiple_ref.output
+
+
+def test_basic(predictions, references, jury, output_basic):
+    scores = jury(predictions=predictions, references=references)
+    assert_almost_equal_dict(actual=scores, desired=output_basic)
+
```
Comment (devrimcavusoglu) [inline tests/jury/metrics/test_bartscore.py:36]: can we add a test case also for segment_scores ?

File: jury/metrics/bartscore/bartscore_for_language_generation.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            "https://raw.githubusercontent.com/neulab/BARTScore/47b8341854e1b8be965b65480ce236b0c2f7543b/bart_score.py"
+        )
+        bartscore_dest = dl_manager.download(bartscore_source)
+        self.external_module_path = bartscore_dest
+        BARTScorer = self._get_external_resource("bart_score", attr="BARTScorer")
+
+        self.scorer = BARTScorer(device=self.device, max_length=self.max_length, checkpoint=model_checkpoint)
+
+        if model_path is not None:
+            model_dest = dl_manager.download(model_path)
```
[the remark this answers is not in the document]
Comment (Sophylax) [reply jury/metrics/bartscore/bartscore_for_language_generation.py:170]: Right now it cannot be a local path, it is extracted from `CHECKPOINT_URLS`, which is a curated list. Supporting any local (or just any in general) path might be something to consider though,

[the remark this answers is not in the document]
Comment (devrimcavusoglu) [reply jury/metrics/bartscore/bartscore_for_language_generation.py:169]: Okay, so let's leave it as a future consideration for now.

**Review by devrimcavusoglu — APPROVED** [review]

**Review by devrimcavusoglu — APPROVED** [review]

**Review by devrimcavusoglu — APPROVED** [review]
Let's set the timeout as 400.

**Pull request merged**